### ISSUE TITLE
`python -m nose` fails for six tests on 0.9.0

### ISSUE THREAD

**Issue by linkmauve** [horazont/aioxmpp#142]
Here are the tracebacks:
```
======================================================================
ERROR: test_verify_recorded_calls_quick_check_for_deferrable_error (tests.test_security_layer.TestHookablePKIXCertificateVerifier)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/tests/test_security_layer.py", line 837, in test_verify_recorded_calls_quick_check_for_deferrable_error
    (self.x509_root, errno, errdepth)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 438, in verify_recorded
    self.hostname_matches = check_x509_hostname(leaf_x509, hostname)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 226, in check_x509_hostname
    cert_structure = extract_python_dict_from_x509(x509)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 173, in extract_python_dict_from_x509
    ("DNS", str(dNSName))
  File "/usr/lib/python3.6/site-packages/pyasn1/type/char.py", line 88, in __str__
    return str(self._value)
  File "/usr/lib/python3.6/site-packages/pyasn1/type/base.py", line 164, in plug
    raise error.PyAsn1Error('Uninitialized ASN.1 value ("%s" attribute looked up)' % name)
pyasn1.error.PyAsn1Error: Uninitialized ASN.1 value ("__str__" attribute looked up)

======================================================================
ERROR: test_verify_recorded_quick_check_None_is_unsure (tests.test_security_layer.TestHookablePKIXCertificateVerifier)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/tests/test_security_layer.py", line 888, in test_verify_recorded_quick_check_None_is_unsure
    (self.x509_root, errno, errdepth)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 438, in verify_recorded
    self.hostname_matches = check_x509_hostname(leaf_x509, hostname)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 226, in check_x509_hostname
    cert_structure = extract_python_dict_from_x509(x509)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 173, in extract_python_dict_from_x509
    ("DNS", str(dNSName))
  File "/usr/lib/python3.6/site-packages/pyasn1/type/char.py", line 88, in __str__
    return str(self._value)
  File "/usr/lib/python3.6/site-packages/pyasn1/type/base.py", line 164, in plug
    raise error.PyAsn1Error('Uninitialized ASN.1 value ("%s" attribute looked up)' % name)
pyasn1.error.PyAsn1Error: Uninitialized ASN.1 value ("__str__" attribute looked up)

======================================================================
ERROR: test_verify_recorded_quick_check_rejects (tests.test_security_layer.TestHookablePKIXCertificateVerifier)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/tests/test_security_layer.py", line 910, in test_verify_recorded_quick_check_rejects
    (self.x509_root, errno, errdepth)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 438, in verify_recorded
    self.hostname_matches = check_x509_hostname(leaf_x509, hostname)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 226, in check_x509_hostname
    cert_structure = extract_python_dict_from_x509(x509)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 173, in extract_python_dict_from_x509
    ("DNS", str(dNSName))
  File "/usr/lib/python3.6/site-packages/pyasn1/type/char.py", line 88, in __str__
    return str(self._value)
  File "/usr/lib/python3.6/site-packages/pyasn1/type/base.py", line 164, in plug
    raise error.PyAsn1Error('Uninitialized ASN.1 value ("%s" attribute looked up)' % name)
pyasn1.error.PyAsn1Error: Uninitialized ASN.1 value ("__str__" attribute looked up)

======================================================================
ERROR: test_verify_recorded_quick_check_unsure (tests.test_security_layer.TestHookablePKIXCertificateVerifier)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/tests/test_security_layer.py", line 861, in test_verify_recorded_quick_check_unsure
    (self.x509_root, errno, errdepth)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 438, in verify_recorded
    self.hostname_matches = check_x509_hostname(leaf_x509, hostname)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 226, in check_x509_hostname
    cert_structure = extract_python_dict_from_x509(x509)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 173, in extract_python_dict_from_x509
    ("DNS", str(dNSName))
  File "/usr/lib/python3.6/site-packages/pyasn1/type/char.py", line 88, in __str__
    return str(self._value)
  File "/usr/lib/python3.6/site-packages/pyasn1/type/base.py", line 164, in plug
    raise error.PyAsn1Error('Uninitialized ASN.1 value ("%s" attribute looked up)' % name)
pyasn1.error.PyAsn1Error: Uninitialized ASN.1 value ("__str__" attribute looked up)

======================================================================
ERROR: test_verify_recorded_rejects_non_deferrable_errors (tests.test_security_layer.TestHookablePKIXCertificateVerifier)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/tests/test_security_layer.py", line 822, in test_verify_recorded_rejects_non_deferrable_errors
    self._test_hardwired(errno, depth)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/tests/test_security_layer.py", line 614, in _test_hardwired
    (self.x509_root, errno, errdepth),
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/tests/test_security_layer.py", line 598, in _test_hardwired_set
    s
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 438, in verify_recorded
    self.hostname_matches = check_x509_hostname(leaf_x509, hostname)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 226, in check_x509_hostname
    cert_structure = extract_python_dict_from_x509(x509)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 173, in extract_python_dict_from_x509
    ("DNS", str(dNSName))
  File "/usr/lib/python3.6/site-packages/pyasn1/type/char.py", line 88, in __str__
    return str(self._value)
  File "/usr/lib/python3.6/site-packages/pyasn1/type/base.py", line 164, in plug
    raise error.PyAsn1Error('Uninitialized ASN.1 value ("%s" attribute looked up)' % name)
pyasn1.error.PyAsn1Error: Uninitialized ASN.1 value ("__str__" attribute looked up)

======================================================================
ERROR: test_zombofant_net (tests.test_security_layer.Testextract_python_dict_from_x509)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/tests/test_security_layer.py", line 226, in test_zombofant_net
    d = security_layer.extract_python_dict_from_x509(x509)
  File "/home/linkmauve/packages/python-aioxmpp/src/aioxmpp-0.9.0/aioxmpp/security_layer.py", line 173, in extract_python_dict_from_x509
    ("DNS", str(dNSName))
  File "/usr/lib/python3.6/site-packages/pyasn1/type/char.py", line 88, in __str__
    return str(self._value)
  File "/usr/lib/python3.6/site-packages/pyasn1/type/base.py", line 164, in plug
    raise error.PyAsn1Error('Uninitialized ASN.1 value ("%s" attribute looked up)' % name)
pyasn1.error.PyAsn1Error: Uninitialized ASN.1 value ("__str__" attribute looked up)

----------------------------------------------------------------------
Ran 3721 tests in 40.492s

FAILED (SKIP=46, errors=6)
```

**Comment by horazont** [comment]
Thanks for the report.

Can you try to cherry-pick or apply 07d3eff? I think that should fix it.

If so, I’ll prepare a bugfix release soon.

**Comment by horazont** [comment]
Fixed with 07d3eff in 0.9.1 bugfix release.
